From f8b49e9c5f40df9447e6adbca57e10f3c44fb153 Mon Sep 17 00:00:00 2001
From: Moss Heim <moss.heim@qt.io>
Date: Thu, 17 Sep 2026 15:20:45 +0200
Subject: [PATCH] [fixup][Backport] CVE-2026-79187: Use after free in WebRTC

Add missing headers for PostCrossThreadTask / CrossThreadCopier

Upstream: https://github.com/qt/qtwebengine-chromium/commit/f8b49e9c5f40df9447e6adbca57e10f3c44fb153
Not in the 6.140.0-rc tarball; drop once the source includes it.

Change-Id: I30b7700a85c91b2a6adeaf4b8035d116264478a0
Reviewed-on: https://codereview.qt-project.org/c/qt/qtwebengine-chromium/+/772028
Reviewed-by: Michal Klocek <michal.klocek@qt.io>
---
 .../peerconnection/webrtc_media_stream_track_adapter_map.cc    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/webrtc_media_stream_track_adapter_map.cc b/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/webrtc_media_stream_track_adapter_map.cc
index b248157b776..9e9f423b21c 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/webrtc_media_stream_track_adapter_map.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/webrtc_media_stream_track_adapter_map.cc
@@ -10,6 +10,9 @@
 #include "base/memory/ptr_util.h"
 #include "base/task/single_thread_task_runner.h"
 #include "third_party/blink/renderer/modules/peerconnection/peer_connection_dependency_factory.h"
+#include "third_party/blink/renderer/platform/scheduler/public/post_cross_thread_task.h"
+#include "third_party/blink/renderer/platform/wtf/cross_thread_copier_base.h"
+#include "third_party/blink/renderer/platform/wtf/cross_thread_functional.h"
 
 namespace blink {
 
